zebra-rpc/src/methods.rs

```diff
@@ -1344,6 +1356,18 @@ where
         }
         .boxed()
     }
+
+    fn stop(&self) -> Result<()> {
+        if self.network.is_regtest() {
+            std::process::exit(0);
```

Comment:
This function doesn't provide a controlled shutdown. We implement a graceful shutdown here: https://github.com/zcashfoundation/zebra/blob/d9fca01d4a5993391dfa591f179bf2217a6a2c75/zebrad/src/application.rs#L513. I think it would be safer to use it instead of direct termination. `zcashd` also uses a controlled shutdown when calling the `stop` RPC.

Reply:
You're right; I considered that as well. However, we don't have access to the application through the RPC methods, so while it's definitely possible, it’s not as straightforward to implement. Since this is currently only for the regtest environment, I believe the direct termination approach should suffice for now. We can implement a proper solution once we remove the regtest limitation. I added a TODO comment to do that.

Reply:
Sounds good.

Reply:
Instead of having a handle to `zebrad`, we could send a signal to the process itself.

- https://docs.rs/nix/0.29.0/nix/sys/signal/fn.raise.html

Reply:
Could we either use a signal or call [`set_shutting_down()`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/shutdown.rs#L27)?